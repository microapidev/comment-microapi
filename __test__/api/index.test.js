const app = require("../../server");
const request = require("supertest");

describe("GET /v1", () => {
  test("Should return status 200 and render Swagger docs", async () => {
    const res = await request(app).get("/v1");
    expect(res.status).toBe(301);
    expect(res.text).toMatchSnapshot();
  });
});

describe("ALL '/xyz' - Invalid routes", () => {
  test("Requests should return status 404 and error message", async () => {
    const res = await request(app).post("/xyz");
    expect(res.status).toBe(404);
    expect(res.body.error).toBe("Oops. The route POST /xyz is not recognised");
  });
});
