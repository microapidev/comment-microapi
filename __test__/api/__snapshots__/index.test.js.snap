// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /v1 Should return status 200 and render Swagger docs 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Redirecting</title>
</head>
<body>
<pre>Redirecting to <a href=\\"/v1/\\">/v1/</a></pre>
</body>
</html>
"
`;
